src/qibocal/auto/task.py

```diff
@@ -243,5 +247,20 @@ def data(self, data: Data):
         self._data = data
         self._data.save(self.datapath)
 
-    def __post_init__(self):
-        self.task = copy.deepcopy(self.task)
+    def update_platform(self, platform: Platform):
+        """Perform update on platform' parameters by looping over qubits or pairs."""
+        if self.task.update:
+            for qubit in self.task.qubits:
+                self.task.operation.update(self.results, platform, qubit)
+
+    def validate(self) -> TaskId:
+        """Check status of completed and handle Failure using handler."""
+        if isinstance(self.status, Failure):
+            try:
+                return self.task.action.handler.id
+            except AttributeError:
+                log.error(
+                    "Stopping execution because of error in validation without handler"
+                )
+                return None
```

Comment:
It is pretty elegant to catch errors in Python, rather than checking, but in this case I would say it's not practical.

Indeed, whatever missing attribute would be raised, even inside the `.handler` it would generate an `AttributeError` indistinguishable from the one you are expecting.

Since in this case it costs very little, I would rather check:
```suggestion
            handler = self.task.action
            if handler is not None:
                return handler.id
            log.error(
                "Stopping execution because of error in validation without handler"
            )
            return None
```